src/vivarium/framework/utilities.py

```diff
@@ -40,30 +42,36 @@ def rate_to_probability(
         The rate to convert to a probability.
     time_scaling_factor
         The time factor in to scale the rates by. This is usually the time step.
+    conversion_type
+        The type of conversion to use. Default is "linear" for a simple multiplcation
+        of rate and time_scaling_factor. Other option is "exponential" which should be
+        used for continuous time event driven models.
 
     Returns
     -------
         An array of floats representing the probability of the converted rates
     """
-    # NOTE: The default behavior for randomness streams is to use a rate that is already
-    # scaled to the time step which is why the default time scaling factor is 1.0.
-    probability = rate * time_scaling_factor
-    # Clip to 1.0 if the probability is greater than 1.0.
-    clip = False
-    exceeds_one = probability > 1.0
-    if isinstance(exceeds_one, bool):
-        if exceeds_one:
-            probability = 1.0
-            clip = True
-    else:
+    if rate_conversion_type == "linear":
+        # NOTE: The default behavior for randomness streams is to use a rate that is already
+        # scaled to the time step which is why the default time scaling factor is 1.0.
+        probability = np.array(rate * time_scaling_factor)
+
+        # Clip to 1.0 if the probability is greater than 1.0.
+        exceeds_one = probability > 1.0
         if exceeds_one.any():
             probability[exceeds_one] = 1.0
-            clip = True
-    if clip:
-        logger.warning(
-            "The rate to probability conversion resulted in a probability greater than 1.0. "
-            "The probability has been clipped to 1.0 and indicates the rate is too high. "
-        )
+            logger.warning(
+                "The rate to probability conversion resulted in a probability greater than 1.0. "
+                "The probability has been clipped to 1.0 and indicates the rate is too high. "
+            )
+    else:
```

Comment:
You should throw a `NotImplemented` error if the user provides a conversion type other than linear or exponential (is exponential the right name for the second conversion strategy?).